### DISCUSSION TITLE
Gulp corrupting imgs after moving

### DISCUSSION THREAD

**Discussion by jpbepu** [gulpjs/gulp#2791] [Help] [answered]
Please help! I have no idea what is happening!
So, recently I tried to move some images from my src directory to the build directory using a standard _src - dest_ task on gulp. The issue that I'm having is that after the task, all of the copied images on the build folder are somehow corrupted. I can't even open them on File Explorer. Why is this happening?

Here is the task that I'm using:

```javascript
function moveIMG(){
    return src('./src/imgs/*')
    .pipe(dest('./build/imgs'))
}
```
The side-by-side of the before and after the move:

![image](https://github.com/gulpjs/gulp/assets/45445802/72222e73-5638-4ace-9de1-8bf5fc198ef6)

Here is the file structure:

```
(project folder)
├── gulpfile.js
├── package-lock.json
├── package.json
├── /node_modules
└── /src
│   ├── index.html
│   ├── /imgs
│   └── /scss
│
└── /build
     ├── index.html
     ├── imgs <----corrupted files
     └── /scss
```

Do I have to set a specific propriety when moving images? Do I HAVE to use imagemin to do this kind of operation?
Thanks is advance!

**Comment by madjacky** [comment]
u must add {encoding: false} to src

**Reply by mdellanave** [reply]
hi @madjacky 
can you please explain the encoding option? or do you know if there's documentation?

In the old version 4, images and fonts were moved without issues and without specifying false value on that option.

thank you

**Reply by UweKeim** [reply, marked as answer]
Updating to version 5 broke my gulpfile.js scripts, too. 😭

Adding `, { encoding: false }` as a second parameter to `gulp.src()` fixed it for me, too.

The [documentation of `src()`](https://gulpjs.com/docs/en/api/src) does _not_ mention the `encoding` option at all.

**Reply by jpbepu** [reply]
Solved! Thanksss

**Reply by kari** [reply]
As to why this is happening, it seems that for some reason, without that option set false in src, Gulp 5 encodes the binary files as UTF-8, and so all invalid UTF-8 code points in the binary data (images, fonts, ...) get converted to invalid codepoint U+FFFD, which is EF BF BD in UTF-8, and this corrupts the files. Someone who knows better Node and Streams can probably pinpoint where this happens.

**Comment by sahksas** [comment]
```diff
function assetsCopy() {
  return gulp
    .src("./src/assets/**", {
+      encoding: false,
    })
    .pipe(gulp.dest("./dist/assets"));
}
```

With the above code change, the issue with the font files (otf, ttf, woff, woff2) not loading has been resolved! Thank you ❤️
